### PR TITLE
Prepare Breeze2 for prime time :) 

### PR DESCRIPTION
**Pull request by potiuk**
This is a review and clean-up for all the parameters and
commands for Breeze2 in order to prepare it for being
used by the contribugors.

There are various small fixes here and there, removal
of duplicated code, refactoring and moving code around
as well as cleanup and review all the parameters used
for all implemented commands.

The parameters, default values and their behaviours were
updated to match "new" life of Breeze rather than old
one.

Some improvements are made to the autocomplete and
click help messages printed.  Full list of choices is
always displayed, parameters are groups according to
their target audience, and they were sorted according
to importance and frequency of use.

Various messages have been colourised according to their
meaning - warnings as yellow, errors as red and
informational messages as bright_blue.

The `dry-run` option has been added to just show what
would have been run without actually running some
potentially "write" commands (read commands are still
executed) so that you can easily verify and manually
copy and execute the commands with option to modify
them before. The `dry_run` and `verbose` options are
now used for all commands.

The "main" command now runs "shell" by default similarly
as the original Breeze.

All "shortcut" parameters have been standardized - i.e
common options (verbose/dry run/help) have one and all
common flags that are likely to be used often have an
assigned shortcute.

The "stop" and "cleanup" command have been added
as they are necessary for average user to complete the
regular usage cycle.

Documentation for all the important methods have been
updated.

This one is based on #22695 so check only last commit

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Comment by potiuk** [comment]
cc: @Bowrna @edithturn -> I have deeeply reviewed, tested and cleaned up the changes.

This is really cool what we have, after I removed some duplications, cleaned up some naming, added some docstrings and reviewed and made sure all the parameters we have, we are 100% ready to replace old breeze and move the old breeze to legacy. This will be the next PR after the #22695  and this one is merged.

My changes were mostly about beautifying things, refactoring/shuffling them around so that they make more sense and making sure we have no duplications. All the rest remained largely as You've implemented it.

FANTASTIC JOB.  Some cool screenshots follow.

**Comment by potiuk** [comment]
Auttocomplete: 

image

Autocomplete 2:

image

Main help with Breeze2: 

image

Build image help:

image

Start-airflow help:

image

Cheatsheet:

image

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -644,7 +644,7 @@ and not packaged together with the core, unless you set ``INSTALL_PROVIDERS_FROM
 variable to ``true``.
 
 In Breeze - which is a development environment, ``INSTALL_PROVIDERS_FROM_SOURCES`` variable is set to true,
-but you can add ``--skip-installing-airflow-providers-from-sources`` flag to Breeze to skip installing providers when
+but you can add ``--install-providers-from-sources=true`` flag to Breeze to skip installing providers when
```
Comment (eladkal) [inline CONTRIBUTING.rst:647]: Shouldn't it be `True`?

**Review by eladkal — APPROVED** [review]
:tada::tada::tada::tada:

File: CONTRIBUTING.rst (hunk 1 above)
Comment (potiuk) [reply CONTRIBUTING.rst:647]: Dockls are not yet updated. I leave it for the switch :) (next PR)

Comment (potiuk) [reply CONTRIBUTING.rst:647]: There will be quite many changes in the docs :)

**Comment by edithturn** [comment]
wow wow, Breeze looks really amazing, @potiuk 👏🏼
Congratulations!

File: dev/breeze/src/airflow_breeze/breeze.py [hunk 2]
```diff
@@ -19,77 +19,414 @@
...
+                    "static-check",
+                ],
+            },
+            {
+                "name": "Configuration & maintenance",
+                "commands": ["cleanup", "setup-autocomplete", "config", "version"],
+            },
+        ]
+    }
+
```
Comment (Bowrna) [inline dev/breeze/src/airflow_breeze/breeze.py:274]: Woah this is really cool :)

**Comment by potiuk** [comment]
Indeed. This week we are going to switch everyone to it :)

**Comment by potiuk** [comment]
And the code is now structured for implementing the rest much better :). I had a lot of fun refactoring and moving parts (with the help of IntelliJ' refactoring features) and it was already very clean and well structured so I mostly just shuffeld things around, made some cleanups and common code extraction and clarified the use of some parameters (removed some of them or simplified them a little. That was mostly it :)

**Comment by potiuk** [comment]
Please take a look at the structure and docs I added - to familarise a bit with the new structure (thought the code is mostly the same, just you will find it in new places sometimes ;)

**Pull request merged**